akd_core/src/utils.rs

```diff
@@ -63,20 +63,27 @@ pub fn get_commitment_nonce(
 }
 
 /// To convert a regular label (arbitrary string of bytes) into a [NodeLabel], we compute the
-/// output as: H(label || stale || version)
+/// output as: H(label || liveness || version)
 ///
 /// Specifically, we concatenate the following together:
 /// - I2OSP(len(label) as u64, label)
 /// - A single byte encoded as 0u8 if "stale", 1u8 if "fresh"
 /// - A u64 representing the version
 /// These are all interpreted as a single byte array and hashed together, with the output
 /// of the hash returned.
-pub(crate) fn get_hash_from_label_input(label: &AkdLabel, stale: bool, version: u64) -> Vec<u8> {
-    let stale_bytes = if stale { &[0u8] } else { &[1u8] };
+pub(crate) fn get_hash_from_label_input(
+    label: &AkdLabel,
+    freshness: VersionFreshness,
+    version: u64,
+) -> Vec<u8> {
+    let freshness_bytes = match freshness {
```

Comment:
should this be part of the struct? Like a deref trait impl on the struct? Just to avoid hardcoding here and keep the logic together. Or add a `to_bytes()` method perhaps?

Reply:
Or perhaps use the actual `#[repr(u8)]` functionality? We can probably cast it so we don't have just hardcoded bytes outside of the struct def

Reply:
Feel free to merge!